webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV2.scala

```diff
@@ -137,36 +139,62 @@ object RouteUtilV2 {
+        case MARKUP_XML      => Validation.succeed(MarkupAsXml)
+        case MARKUP_STANDOFF => Validation.succeed(MarkupAsStandoff)
+        case _               => Validation.fail(BadRequestException(s"Unrecognised standoff rendering: $standoffRenderingName"))
       }
 
     val params: Map[String, String] = requestContext.request.uri.query().toMap
 
     params.get(MARKUP_PARAM) match {
-      case Some(schemaParam) => Some(nameToStandoffRendering(schemaParam))
+      case Some(schemaParam) => nameToStandoffRendering(schemaParam).map(Some(_))
 
       case None =>
-        requestContext.request.headers.find(_.lowercaseName == MARKUP_HEADER).map { header =>
-          nameToStandoffRendering(header.value)
-        }
+        requestContext.request.headers
+          .find(_.lowercaseName == MARKUP_HEADER)
+          .map(_.value)
+          .fold[Validation[Throwable, Option[MarkupRendering]]](Validation.succeed(None))(
+            nameToStandoffRendering(_).map(Some(_))
+          )
     }
   }
 
-  private def getJsonLDRendering(requestContext: RequestContext): Option[JsonLDRendering] = {
-    def nameToJsonLDRendering(jsonLDRenderingName: String): JsonLDRendering =
-      jsonLDRenderingName match {
-        case JSON_LD_RENDERING_FLAT         => FlatJsonLD
-        case JSON_LD_RENDERING_HIERARCHICAL => HierarchicalJsonLD
-        case _                              => throw BadRequestException(s"Unrecognised JSON-LD rendering: $jsonLDRenderingName")
-      }
+  /**
+   * Gets the type of standoff rendering that should be used when returning text with standoff.
+   * The name of the standoff rendering can be specified either in the HTTP header [[MARKUP_HEADER]]
+   * or in the URL parameter [[MARKUP_PARAM]]. If no rendering is specified in the request, the
+   * default of [[MarkupAsXml]] is returned.
+   *
+   * @param requestContext the akka-http [[RequestContext]].
+   * @return the specified standoff rendering, or [[MarkupAsXml]] if no rendering was specified
+   *         in the request.
+   */
+  @deprecated("Use getStandoffRendering(requestContext: RequestContext) instead")
+  private def getStandoffRenderingUnsafe(requestContext: RequestContext): Option[MarkupRendering] =
+    getStandoffRendering(requestContext).fold(
+      errors => throw errors.head,
+      markupRendering => markupRendering
+    )
+
+  @deprecated("use getJsonLDRendering() instead")
+  private def getJsonLDRenderingUnsafe(requestContext: RequestContext): Option[JsonLDRendering] =
+    getJsonLDRendering(requestContext).fold(
+      errors => throw errors.head,
+      jsonLDRendering => jsonLDRendering
+    )
 
-    requestContext.request.headers.find(_.lowercaseName == JSON_LD_RENDERING_HEADER).map { header =>
-      nameToJsonLDRendering(header.value)
+  private def getJsonLDRendering(requestContext: RequestContext): Validation[Throwable, Option[JsonLDRendering]] = {
```

Comment:
```suggestion
  private def getJsonLDRendering(requestContext: RequestContext): Validation[BadRequestException, Option[JsonLDRendering]] = {
```

narrow the type please.